src/NumericalAlgorithms/DiscontinuousGalerkin/Actions/InitializeDomain.hpp

```diff
@@ -40,76 +40,90 @@ class ConstGlobalCache;
 
 namespace dg {
 namespace Actions {
-/// \ingroup InitializationGroup
-/// \brief Initialize items related to the basic structure of the Domain
-///
-/// DataBox changes:
-/// - Adds:
-///   * `Tags::Mesh<Dim>`
-///   * `Tags::Element<Dim>`
-///   * `Tags::ElementMap<Dim>`
-///   * `Tags::LogicalCoordinates<Dim>`
-///   * `Tags::MappedCoordinates`
-///   * `Tags::InverseJacobian`
-///   * `Tags::MinimumGridSpacing<Dim, Frame::Inertial>>`
-/// - Removes: nothing
-/// - Modifies: nothing
-template <size_t Dim>
+/*!
+ * \ingroup InitializationGroup
+ * \brief Initialize items related to the basic structure of the element
+ *
+ * DataBox:
+ * - Uses:
+ *   - `Tags::Domain<Dim, Frame::Inertial>`
+ *   - `Tags::InitialExtents<Dim>`
+ * - Adds:
+ *   - `Tags::Mesh<Dim>`
+ *   - `Tags::Element<Dim>`
+ *   - `Tags::ElementMap<Dim, Frame::Inertial>`
+ *   - `Tags::Coordinates<Dim, Frame::Logical>`
+ *   - `Tags::Coordinates<Dim, Frame::Inertial>`
+ *   - `Tags::InverseJacobian<
+ *   Tags::ElementMap<Dim>, Tags::Coordinates<Dim, Frame::Logical>>`
+ *   - `Tags::MinimumGridSpacing<Dim, Frame::Inertial>>`
+ * - Removes: nothing
+ * - Modifies: nothing
+ */
+template <size_t Dim, ::Initialization::MergePolicy MergePolicy =
+                          ::Initialization::MergePolicy::Error>
```

Comment:
What is `MergePolicy` and why is it being added?

Reply:
It was introduced in #1579 for `::Initialization::merge_into_databox`, which is now used in all initialisers. It allows multiple initialisers to add the same tags (see https://spectre-code.org/group__initializationgroup#gafc4ef63b38e015766414946b0b5523b5). The `MergePolicy` decides what to do with conflicts. It also makes dealing with DataBox chains in action lists easier. I suggest to generally template initializers on the `MergePolicy` and default it to `Error` so that this mechanism can be used.

Reply:
This domain initialiser is particularly tricky, because `ElementMap`s are not comparable. So one can't just have an action list `tmpl::list<InitializeDomain<Dim>>` by itself: The DataBox that comes out of an iteration of this list contains a `Tags::ElementMap`, but it must be compatible with feeding it back into the start of the list. `MergePolicy::Error` fails here, because when the second iteration tries to add an `ElementMap` again it can't check if it's equal to the existing one. Using `MergePolicy::IgnoreIncomparable` would fix this.

Reply:
@nilsleiffischer can you give a concrete example of when one would need to use a `MergePolicy` because I really, really dislike it and strongly disagree with generally templating initializers with it...

Reply:
An example would be re-initializing the linear solver: In an executable that does multiple linear solves, one sets `MergePolicy::Overwrite` for the re-initialization.

If we don't template initializers on the `MergePolicy`, then there is no way to use this functionality of `merge_into_databox`, since the function will only ever be used in initialisers.

Reply:
I don't like this functionality in `merge_into_databox`, it seems like a hack to me.  I don't think initialization actions should be used outside of the initialization phase.  Why can't you just mutate the existing items when re-initializing the linear solver?

Reply:
I could, but using the same initializer for re-initialization seems quite elegant to me and reduces code duplication. But let's talk about the linear solver when the time comes. I really need to move forward with these PRs. I removed the `MergePolicy` here for now and instead work around the issue in the test. Please check the fixup :)